packages/zwave-js/src/lib/controller/ReplaceFailedNodeRequest.ts

```diff
@@ -0,0 +1,143 @@
+import type { Driver } from "../driver/Driver";
+import {
+	FunctionType,
+	MessagePriority,
+	MessageType,
+} from "../message/Constants";
+import {
+	expectedCallback,
+	expectedResponse,
+	gotDeserializationOptions,
+	Message,
+	MessageBaseOptions,
+	MessageDeserializationOptions,
+	MessageOptions,
+	messageTypes,
+	priority,
+} from "../message/Message";
+import type { SuccessIndicator } from "../message/SuccessIndicator";
+
+export enum ReplaceFailedNodeStartFlags {
+	OK = 0,
+	/** The replacing process was aborted because the controller  is not the primary one */
+	NotPrimaryController = 1 << 1,
+	/** The replacing process was aborted because no call back function is used */
+	NoCallbackFunction = 1 << 2,
+	/** The replacing process aborted because the node was node found */
+	NodeNotFound = 1 << 3,
+	/** The replacing process is busy */
+	ReplaceProcessBusy = 1 << 4,
+	/** The replacing process could not be started*/
+	ReplaceFailed = 1 << 5,
+}
+
+export enum ReplaceFailedNodeStatus {
+	/* ZW_ReplaceFailedNode and ZW_ReplaceFailedNode callback status definitions */
+	/** The node is working properly (removed from the failed nodes list ). Replace process stopped */
+	NodeOK = 0,
+
+	/* ZW_ReplaceFailedNode callback status definitions */
+	/** The failed node is ready to be replaced
+        and controller is ready to add new node
+        with the nodeID of the failed node.
+        Meaning that the new node must now
+        emit a nodeinformation frame to be
+        included. */
+	FailedNodeReplace = 1,
+	/** The failed node has been replaced. */
+	FailedNodeReplaceDone = 2,
+
+	/** The replace process has failed */
+	FailedNodeReplaceFailed = 3,
+}
+
+@messageTypes(MessageType.Request, FunctionType.ReplaceFailedNode)
+@priority(MessagePriority.Controller)
+export class ReplaceFailedNodeRequestBase extends Message {
+	public constructor(driver: Driver, options: MessageOptions) {
+		if (
+			gotDeserializationOptions(options) &&
+			(new.target as any) !== ReplaceFailedNodeRequestStatusReport
+		) {
+			return new ReplaceFailedNodeRequestStatusReport(driver, options);
+		}
+		super(driver, options);
+	}
+}
+
+interface ReplaceFailedNodeRequestOptions extends MessageBaseOptions {
+	// This must not be called nodeId or rejectAllTransactions may reject the request
+	failedNodeId: number;
+	includeNonSecure: boolean;
+}
+
+@expectedResponse(FunctionType.ReplaceFailedNode)
+@expectedCallback(FunctionType.ReplaceFailedNode)
```

Comment:
If the controller actually sends multiple callbacks, you cannot use `@expectedCallback` to define it here.
See AddNodeToNetworkRequest.ts, line 51
This is okay, since you're handling these messages with the handler function you added to Controller.ts

Reply:
Removed that